internal/controller/pkg/manager/revisioner.go

```diff
@@ -76,6 +77,13 @@ func (r *PackageRevisioner) Revision(ctx context.Context, p v1.Package) (string,
 			return p.GetCurrentRevision(), nil
 		}
 	}
+
+	// NOTE(ezgidemirel): Package source may contain a tag or a digest. If it contains a digest,
+	// we need to convert the format from `:sha256:digest` to `@sha256:digest`
+	if strings.Contains(p.GetSource(), "sha256") {
```

Comment:
can you use here go-container registry - something around

```
	// If the reference is a digest, convert it to the canonical form
	if digest, ok := ref.(name.Digest); ok {
		// Convert the digest to its canonical form, which uses the "@" format
		return digest.String(), nil
	}
```

Reply:
Any reason not to use the canonical `@` syntax in the package metadata? That would avoid the need for this conversion altogether.

Reply:
@haarchri thanks for the suggestion. I converted the constraint to hash in the resolver/reconciler.go

Reply:
> Any reason not to use the canonical `@` syntax in the package metadata? That would avoid the need for this conversion altogether.

If the parent package uses a digest, it should come with the canonical `@` syntax like below: 
```
apiVersion: pkg.crossplane.io/v1
kind: Configuration
metadata:
  name: configuration-getting-started
spec:
  package: xpkg.upbound.io/upbound/configuration-getting-started@sha256:869bca254eb12f2d5d9681ac7186f13e67e2984e68f18e6195492d37e4fb42a2
  ignoreCrossplaneConstraints: true
  packagePullPolicy: Always
```

But if we use a dependency with digest, we should format the package name and constraint with the corresponding format. That part is handled in resolver/reconciler.go. This replace code was unnecessary, removed.